test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -101,6 +131,63 @@ func InitManagementClusterAndWatchControllerLogs(ctx context.Context, input Init
 	}
 }
 
+// UpgradeManagementClusterAndWaitInput is the input type for UpgradeManagementClusterAndWait.
+type UpgradeManagementClusterAndWaitInput struct {
+	ClusterProxy         framework.ClusterProxy
+	ClusterctlConfigPath string
+	ManagementGroup      string
+	Contract             string
+	LogFolder            string
+}
+
+// UpgradeManagementClusterAndWait upgrades provider a management cluster using clusterctl, and waits for the cluster to be ready.
+func UpgradeManagementClusterAndWait(ctx context.Context, input UpgradeManagementClusterAndWaitInput, intervals ...interface{}) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for UpgradeManagementClusterAndWait")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling UpgradeManagementClusterAndWait")
+	Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling UpgradeManagementClusterAndWait")
+	Expect(input.ManagementGroup).ToNot(BeEmpty(), "Invalid argument. input.ManagementGroup can't be empty when calling UpgradeManagementClusterAndWait")
+	Expect(input.Contract).ToNot(BeEmpty(), "Invalid argument. input.Contract can't be empty when calling UpgradeManagementClusterAndWait")
+	Expect(os.MkdirAll(input.LogFolder, 0755)).To(Succeed(), "Invalid argument. input.LogFolder can't be created for UpgradeManagementClusterAndWait")
+
+	Upgrade(ctx, UpgradeInput{
+		ClusterctlConfigPath: input.ClusterctlConfigPath,
+		KubeconfigPath:       input.ClusterProxy.GetKubeconfigPath(),
+		ManagementGroup:      input.ManagementGroup,
+		Contract:             input.Contract,
+		LogFolder:            input.LogFolder,
+	})
+
+	client := input.ClusterProxy.GetClient()
+
+	log.Logf("Waiting for provider controllers to be running")
+	controllersDeployments := framework.GetControllerDeployments(ctx, framework.GetControllerDeploymentsInput{
+		Lister:            client,
+		ExcludeNamespaces: []string{"capi-webhook-system"}, // this namespace has been dropped in v1alpha4; this ensures we are not waiting for deployments being deleted as part of the upgrade process
```

Comment:
Yes, 
it is a problem here because if old deployments are not yet deleted at this stage, then the test will try to capture logs from them, but they are being deleted

Reply:
If we want, we could verify that the old webhook system namespace is in fact deleted, but up to you if we want to cover that here.

Reply:
yes, that's another option, but this allows us to move on while things are deleted (instead of waiting for deletion)